static/js/settings_bots.js

```diff
@@ -493,7 +493,7 @@ export function set_up() {
                         const service_interface = $("#edit_service_interface :selected").val();
                         formData.append("service_payload_url", JSON.stringify(service_payload_url));
                         formData.append("service_interface", service_interface);
-                    } else if (type === EMBEDDED_BOT_TYPE) {
+                    } else if (type === EMBEDDED_BOT_TYPE && service !== undefined) {
```

Comment:
If missing this code was causing a 500, there's probably additionally a server bug too; can you investigate that?

Reply:
To explain clearly, we were passing `config_data` as `{}` even for bots that do not have any config info. So, in the server, the below lines call `do_update_bot_config_data` which calls `get_bot_config` and this raises error since there is no BotConfigData object for this bot.

```
if config_data is not None:
        do_update_bot_config_data(bot, config_data)
```

```
def get_bot_config(bot_profile: UserProfile) -> Dict[str, str]:
    entries = BotConfigData.objects.filter(bot_profile=bot_profile)
    if not entries:
        raise ConfigError("No config data available.")
    return {entry.key: entry.value for entry in entries}
```

Can we also check for `config_data != {}` in the server?